docs/architecture/actor_system.md

```diff
@@ -0,0 +1,117 @@
+# Rally Actor System
+
+At its heart, Rally is a distributed system. It has been designed that way to
+allow using multiple load drivers in the same benchmark, to ensure that Rally
+is never a bottleneck. In the vast majority of cases, using a powerful load
```

Comment:
Another reason (the first one, IIRC) is that distributing Rally also provides an easy way to handle multiple target hosts and ensure that they all build/install/start the required ES version via the `race` subcommand (this is for example the mode used on the Hetzner nightly benchmarks).
Later on we got the dedicated/decoupled `esrally build/install/start/stop` subcommands (which we use e.g. in esbench) and these are decoupled from the actor system.

Reply:
The current formulation isn't correct then, however my understanding is that this is achievable without a full-fledged actor system, while we could easily end up re-implementing Thespian badly to support multiple load drivers.

Reply:
> however my understanding is that this is achievable without a full-fledged actor system, while we could easily end up re-implementing Thespian badly to support multiple load drivers.

I mostly agree. I don't think the risk of supporting >1 load drivers without the actor system would necessarily result in re-implementing Thespian, but as always, the devil is in the details.

Reply:
I've changed the wording slightly in https://github.com/elastic/rally/pull/1748/commits/ac1aec8d28bbfb0623dd1e579a241f281344eafb.